2015-Fall/mhartvig/Calculator/Calculator/ViewController.swift

```diff
@@ -0,0 +1,98 @@
+//
+//  ViewController.swift
+//  Calculator
+//
+//  Created by Martin Hartvig on 04/09/15.
+//  Copyright (c) 2015 Martin Hartvig. All rights reserved.
+//
+
+import UIKit
+
+class ViewController: UIViewController {
+    
+    
+    @IBOutlet weak var dotButton: UIButton!
+    @IBOutlet weak var display: UILabel!
+    @IBOutlet weak var historyLabel: UILabel!
+    
+    var isUserTyping: Bool = false;
+    var isNumberADecimal: Bool = false;
+    
+    let brain = CalculatorBrian()
+    
+    override func viewDidLoad() {
+        historyLabel.text = ""
+    }
+    
+    var displayValue: Double? {
+        get {
+            if let value = NSNumberFormatter().numberFromString(display.text!) {
+                return value.doubleValue
+            }
+            return nil
+        }
+        
+        set {
+            if let v = newValue {
+                display.text = "\(v)"
+            } else {
+                display.text = "ERROR"
+            }
+        }
+    }
+    @IBAction func operate(sender: UIButton) {
+        if isUserTyping {
+            enterPress()
+        }
+        
+        if let operation = sender.currentTitle {
+            displayValue = brain.performOperation(operation)
+            historyLabel.text = brain.history()
+        }
+        
+    }
+   
+    
+    @IBAction func enterPress() {
+        isUserTyping = false
+        isNumberADecimal = false
+        if let v = displayValue {
+            brain.pushValue(v)
+        }
+        displayValue = 0
+        historyLabel.text = brain.history()
+    }
+    
+    
+    
+    @IBAction func numberPress(sender: UIButton) {
+        let number = sender.titleLabel!.text!;
+        
+        if (isUserTyping) {
+            display.text = display.text! + number;
+        } else {
+            display.text = number;
+            isUserTyping = true;
+        }
+    }
+    @IBAction func Clear(sender: AnyObject) {
```

Comment:
True and good point. However, this method signature is very common as it often makes sense that protocols specify signatures where the sender object is passed along. It also makes it fast to see what kind of control invokes the action. I always keep the argument as done here, regardless of using the sender or not.